.travis.yml

```diff
@@ -0,0 +1,21 @@
+dist: trusty
+language: csharp
+solution: Banshee.sln
+
+addons:
+  apt:
+    packages:
+      - libgtk3.0-cil
+      - libdbus2.0-cil
+      - libtaglib2.1-cil
+      - libmono-addins0.2-cil
+      - libdbus-glib2.0-cil
+      - libgudev3.0-cil
+      - libmono-zeroconf1.0-cil
+      - libgconf2.0-cil
+env:
+  - BUILD_CONFIGURATION=Release
+  - BUILD_CONFIGURATION=Debug
+
+script:
+  - msbuild /p:Configuration=$BUILD_CONFIGURATION Banshee.sln
```

Comment:
hmm, I hadn't thought much about CI when I did the autotools changes in 1e4ea0988358c0760be79d3ad5a725017342e811 (switches for everything, disable all by default)

I think we should stick to using them for building in a CI chain - although it might be more natural to use a solution file here.

I tried master's autotools setup and had to use ./autoregen.sh instead of the generic autoreconf -fvi though and I think if we add part of the above sha1's patch (the M4_DIRS change in configure.ac) then we can get it working, we'll have to add a bunch of --enable/disable switches.